cmd/status/status.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/jaegertracing/jaeger/cmd/flags"
+	"github.com/jaegertracing/jaeger/ports"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// Command for running status check against the AdminPort.
+func Command(v *viper.Viper, adminPort int) *cobra.Command {
+	adminServer := flags.NewAdminServer(ports.PortToHostPort(adminPort))
+	adminURL := ""
+	c := &cobra.Command{
+		Use:   "status",
+		Short: "Print the status.",
+		Long:  `Prints admin status information, exit non-zero on any error.`,
+		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			rootCmd := cmd
+			for rootCmd.Parent() != nil {
+				rootCmd = rootCmd.Parent()
+			}
+			// FIXME: arg overrides aren't working, adminServer doesn't seem to see the current flagSet. Why?
+			adminURL, err = adminServer.URL()
+			if err != nil {
+				return fmt.Errorf("no default admin port available for %s: %w", cmd.Name(), err)
+			}
+			return nil
+		},
+		// FIXME: Normally I'd want to return an Error and let the exit handler do the os.Exit.
+		// But it's triggering --help output everytime and I don't know why.
+		Run: func(cmd *cobra.Command, args []string) {
+			adminURL, _ = adminServer.URL()
+			resp, err := http.Get(adminURL)
```

Comment:
I am not sure that this command needs to go through HTTP endpoint. Could we not pass the healthcheck object to it that all `main()`s instantiate?

Reply:
The purpose of the command is to provide an easy hook to make status checking possible in something like AWS ECS where only CMD are permitted and the Jaeger containers are running in empty environments (which is how they are distributed).
In essence enabling `docker exec -it jaeger-agent status` to report the currently running service's status. Since it's running as a different PID, I'm not clear how I could pass the healthcheck object to it and get that behavior. I did play around passing the object through, but it didn't expose the listening port and I still needed a url. If you think modifying the admin healthcheck interface is better, I'd be happy to do it. I just couldn't grasp how & when flags are being propagated.

Reply:
oh, you're right, never mind.